### PR TITLE
Add `parents` field to project structure #8154

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/CreateProjectHandler.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/SetProjectParentParamsJson.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/SetProjectParentHandler.java  3
- modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/SetProjectParentHandlerTest.java  1
- modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectParentMapper.java  3
- modules/lib/lib-project/src/test/resources/test/SetProjectParentHandlerTest.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5705878&bid=18675414)

File: modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectParentMapper.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Override
+    public void serialize( final MapGenerator gen )
+    {
+        doSerialize( gen );
+    }
+
+    private void doSerialize( final MapGenerator gen )
+    {
+//        gen.map();
```
Comment (anatol-sialitski) [inline modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectParentMapper.java:25]: remove commented code

File: modules/lib/lib-project/src/main/resources/lib/xp/project.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ exports.create = function (params) {
     bean.language = nullOrValue(params.language);
     bean.permissions = __.toScriptValue(params.permissions);
     bean.readAccess = __.toScriptValue(params.readAccess);
+    bean.parent = nullOrValue(params.parent);
```
Comment (anatol-sialitski) [inline modules/lib/lib-project/src/main/resources/lib/xp/project.js:55]: there is the `__.nullOrValue` method in Java bridge.

File: modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectConstants.java [hunk 3]
```diff
@@ -21,13 +21,7 @@
 
     public static final String PROJECT_ICON_PROPERTY = "icon";
 
-    public static final String PROJECT_ACCESS_LEVEL_OWNER_PROPERTY = "owner";
```
[the remark this answers is not in the document]
Comment (vbradnitski) [reply modules/core/core-api/src/main/java/com/enonic/xp/project/ProjectConstants.java:24]: Has anyone started to use them?

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/CreateProjectParamsJson.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyProjectParamsJson.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5705878&bid=18675414)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java  2
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyProjectParamsJson.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/CreateProjectParamsJson.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5705878&bid=18675414)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/CreateProjectParamsJson.java  3
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ModifyProjectParamsJson.java  1
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/project/json/ProjectJson.java  2
         

Clones removed
==============
+ modules/lib/lib-project/src/main/resources/lib/xp/project.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5705878&bid=18675414)

**Pull request merged**